### PR TITLE
feat(Response): Get status code by methods

### PR DESCRIPTION
**Pull request by booleanrecep**
#### Description
Some of most important status codes can be retrieved via method call.
### List of added status codes
200, 301, 307, 400, 401, 403, 404, 500, 502, 503

`Before`
```go
app.Get("/", func(res *minima.Response, req *minima.Request) {
	res.Status(200).Send("Hello World")
})

app.Get("/", func(res *minima.Response, req *minima.Request) {
	res.Status(404).Send("Hello World")
})
```
`Now`
```go
app.Get("/", func(res *minima.Response, req *minima.Request) {
	res.Status(res.OK()).Send("Hello World")
})

app.Get("/", func(res *minima.Response, req *minima.Request) {
	res.Status(res.NotFound()).Send("Hello World")
})
```

### CODE REVIEW

**Comment by apoorvcodes** [comment]
Maybe instead of returning the status code it explicitly keeps the status

```go
res.Status(res.Ok()) 
```
This is a good implementation but something like this can be better and more simple

```go
res.Ok() //sets 200 directly
```

File: response.go [hunk 1]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
 	http.SetCookie(res.ref, cookie)
 	return res
 }
+
+/**
+@info Set status code as 200
+@returns 200
```
Comment (megatank58) [inline response.go:198]:
```suggestion
@returns {200}
```

File: response.go [hunk 2]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+*/
+func (res *Response) OK() int {
+	return statusCodes["OK"]
+}
+
+/**
+@info Set status code as 301
+@returns 301
```
Comment (megatank58) [inline response.go:206]:
```suggestion
@returns {301}
```

File: response.go [hunk 3]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+*/
+func (res *Response) MovedPermanently() int {
+	return statusCodes["Moved Permanently"]
+}
+
+/**
+@info Set status code as 307
+@returns 307
```
Comment (megatank58) [inline response.go:214]:
```suggestion
@returns {307}
```

File: response.go [hunk 4]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+*/
+func (res *Response) TemporaryRedirect() int {
+	return statusCodes["Temporary Redirect"]
+}
+
+/**
+@info Set status code as 400
+@returns 400
```
Comment (megatank58) [inline response.go:222]:
```suggestion
@returns {400}
```

File: response.go [hunk 5]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+@info Set status code as 502
+@returns 502
+*/
+func (res *Response) BadGateway() int {
+	return statusCodes["Bad Gateway"]
+}
+
+/**
+@info Set status code as 503
+@returns 503
```
Comment (megatank58) [inline response.go:270]:
```suggestion
@returns {503}
```

File: response.go [hunk 6]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+@info Set status code as 500
+@returns 500
+*/
+func (res *Response) InternalServerError() int {
+	return statusCodes["Internal Server Error"]
+}
+
+/**
+@info Set status code as 502
+@returns 502
```
Comment (megatank58) [inline response.go:262]:
```suggestion
@returns {502}
```

File: response.go [hunk 7]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+@info Set status code as 404
+@returns 404
+*/
+func (res *Response) NotFound() int {
+	return statusCodes["NOT FOUND"]
+}
+
+/**
+@info Set status code as 500
+@returns 500
```
Comment (megatank58) [inline response.go:254]:
```suggestion
@returns {500}
```

File: response.go [hunk 8]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+@info Set status code as 403
+@returns 403
+*/
+func (res *Response) Forbidden() int {
+	return statusCodes["Forbidden"]
+}
+
+/**
+@info Set status code as 404
+@returns 404
```
Comment (megatank58) [inline response.go:246]:
```suggestion
@returns {404}
```

File: response.go [hunk 9]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+@info Set status code as 401
+@returns 401
+*/
+func (res *Response) Unauthorized() int {
+	return statusCodes["Unauthorized"]
+}
+
+/**
+@info Set status code as 403
+@returns 403
```
Comment (megatank58) [inline response.go:238]:
```suggestion
@returns {403}
```

File: response.go [hunk 10]
```diff
@@ -192,3 +192,83 @@ func (res *Response) ClearCookie(cookie *http.Cookie) *Response {
...
+@info Set status code as 400
+@returns 400
+*/
+func (res *Response) BadRequest() int {
+	return statusCodes["Bad Request"]
+}
+
+/**
+@info Set status code as 401
+@returns 401
```
Comment (megatank58) [inline response.go:230]:
```suggestion
@returns {401}
```

**Review by megatank58 — CHANGES_REQUESTED** [review]
Just some nits in the documentation comments rest looks fine to me

**Review by apoorvcodes — CHANGES_REQUESTED** [review]
Looks decent kindly do the changes that are asked by @megatank58

**Comment by megatank58** [comment]
LGTM!

**Review by megatank58 — APPROVED** [review]

**Comment by booleanrecep** [comment]
> Maybe instead of returning the status code it explicitly keeps the status
> 
> ```go
> res.Status(res.Ok()) 
> ```
> 
> This is a good implementation but something like this can be better and more simple
> 
> ```go
> res.Ok() //sets 200 directly
> ```

That looks nicer really. Now it's possible to do so:
```go
res.OK().Send("Hello  World")
``` 
But one thing I couldn't figure out is that:
```go
res.OK().Send("Hello").NotFound().Send(" World")
```
They both give the same response and even if  NotFound() has no affect, I think developer should be restricted to be able to write such a chained status method call. But I couldn't find a solution how to disable it.

**Comment by apoorvcodes** [comment]
It won't happen cos go doesn't allow to write status header after content is written to the page

**Comment by apoorvcodes** [comment]
So i dont think so you gotta be worried about it, great work btw :)

**Pull request merged**